pandas/core/ops.py

```diff
@@ -1435,6 +1436,82 @@ def add_flex_arithmetic_methods(cls):
     add_methods(cls, new_methods=new_methods)
 
 
+# ----------------------------------------------------------------------
+# Unary
+
+def _unary_method(cls, op, special=True):
+    op_name = _get_op_name(op, special)
+
+    def f(self):
+        if self.ndim > 2:
+            with np.errstate(all='ignore'):
+                # apply op at once based on current impl
+                new_data = op(self._values)
+            result = self._constructor(new_data, self.items,
+                                       self.major_axis, self.minor_axis)
+            return result.__finalize__(self)
+        elif self.ndim == 2:
+            new_data = {i: op(self.iloc[:, i])
+                        for i in range(len(self.columns))}
+            result = self._constructor(new_data, index=self.index, copy=False)
+            # Pin columns instead of passing to constructor for compat with
+            # non-unique columns case
+            result.columns = self.columns
+            return result.__finalize__(self)
+        else:
+            with np.errstate(all='ignore'):
+                new_data = op(self._values)
+            return self._constructor(new_data, name=self.name,
+                                     index=self.index).__finalize__(self)
+
+    f.__name__ = op_name
+
+    return f
+
+
+def pos(values):
```

Comment:
Should these be added to `ops._get_opstr` and  `ops._op_description`?

Reply:
I understand `_get_opstr` is required to pass expression to `NumExpr`, and `_op_description` is to generate flex methods docstring. Is unary methods needs them?